main/lsp/LSPConfiguration.cc

```diff
@@ -99,7 +99,14 @@ core::Loc LSPConfiguration::lspPos2Loc(const core::FileRef fref, const Position
     core::Loc::Detail reqPos;
     reqPos.line = pos.line + 1;
     reqPos.column = pos.character + 1;
-    auto offset = core::Loc::pos2Offset(fref.data(gs), reqPos);
```

Comment:
does this ~reimplement pos2Offset?
could we make it return `optional` instead?

Reply:
Good suggestion! In the process of refactoring i uncovered an ENFORCE failure that only manifests for files with a class def on the last line of a file with no line break at the end of the file. We don't crash on that anymore!